cmd/gcp-routes-controller/run.go

```diff
@@ -196,3 +196,142 @@ func (sl *healthTracker) OnComplete(state *health.State) {
 		}
 	}
 }
+
+func getRoutes() (map[string]string, error) {
+	routes := make(map[string]string)
+	const netPath = "network-interfaces/"
+	ifs, err := getInstances(netPath)
+	if err != nil {
+		return nil, err
+	}
+	for _, vif := range ifs {
+		macs, err := getInstances(netPath + vif + "mac")
+		if err != nil {
+			return nil, err
+		}
+		hwAddr := macs[0]
```

Comment:
is this safe? The fields come from a google URL. It can break and return whatever.

Reply:
It's done with curl in the original script, see https://github.com/LorbusChris/installer/blob/0fbddf1cd735a29282415aa0dae370bad9af5462/data/data/bootstrap/gcp/files/usr/local/bin/gcp-routes.sh#L57

Suggestions to improve this are welcome ofc